src/plugins/workspace/public/components/workspace_detail/select_data_source_panel.tsx

```diff
@@ -42,11 +45,64 @@ export const SelectDataSourceDetailPanel = ({
   currentWorkspace,
 }: SelectDataSourcePanelProps) => {
   const {
-    services: { notifications, workspaceClient },
+    services: { notifications, workspaceClient, http },
   } = useOpenSearchDashboards<{ CoreStart: CoreStart; workspaceClient: WorkspaceClient }>();
   const { formData, setSelectedDataSources } = useWorkspaceFormContext();
   const [isLoading, setIsLoading] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
+  const [dataSourceConnections, setDataSourceConnections] = useState<DataSourceConnection[]>([]);
+  const [toggleIdSelected, setToggleIdSelected] = useState('all');
+
+  const fetchDQC = useCallback(async () => {
```

Comment:
Shall we separate this DQC fetch logic in a standalone hook?

Reply:
Sure, I have separate this function to `src/plugins/workspace/public/hooks.ts`.